src/git-command-manager.ts

```diff
@@ -154,6 +163,27 @@ class GitCommandManager {
     return result
   }
 
+  async sparseCheckout(sparseCheckout: string[]): Promise<void> {
+    await this.execGit(['sparse-checkout', 'set', ...sparseCheckout])
```

Comment:
do we need to have extra check to make sure git is greater than certain version that support `sparse-checkout`?

Reply:
Hey @TingluoHuang [git sparse-checkout](https://git-scm.com/docs/git-sparse-checkout) is supported since the git version `2.26.0`  and the current version on `actions/checkout` of git is `2.40.1` so I don't think this is something necessary 🤔

Reply:
The current Git version on _hosted runners_ is v2.40.1. But `actions/checkout` also supports self-hosted runners where users are often not at liberty to bump the Git version, e.g. because they're stuck on old Ubuntu LTS versions which simply refuse to move the needle.

I added https://github.com/actions/checkout/pull/1369/commits/c4602b42ee2a5f835f91eb29a5f4e55c060afa82 to address this.

Reply:
Uu nice! I understand perfect ✨